docs/usage/self-hosted-configuration.md

```diff
@@ -65,82 +64,179 @@ e.g.
 
 ## autodiscover
 
-Be cautious when using this option - it will run Renovate over _every_ repository that the bot account has access to.
-To filter this list, use `autodiscoverFilter`.
+When you enable `autodiscover`, by default, Renovate will run on _every_ repository that the bot account can access.
+This is probably not what you want.
+Use the `autodiscoverFilter` option to limit the bot to only the wanted repositories.
 
 ## autodiscoverFilter
 
-A [minimatch](https://www.npmjs.com/package/minimatch) glob-style pattern for filtering `autodiscover`ed repositories. Ex: `project/*`
+You can use this option to filter the list of repositories that the Renovate bot account can access trough `autodiscover`.
+It takes a [minimatch](https://www.npmjs.com/package/minimatch) glob-style pattern.
+
+e.g.
+
+```json
+{
+  "autodiscoverFilter": "project/*"
+}
+```
 
 ## baseDir
 
-Configure this directory if you want to change which directory Renovate uses for storing data.
-If left unconfigured, it will typically be a temporary directory like `/tmp/renovate/`.
+By default Renovate uses a temporary directory like `/tmp/renovate` to store its data.
+You can override this default with the `baseDir` option.
+
+e.g.
+
+```json
+{
+  "baseDir": "/my-own-different-temporary-folder"
+}
+```
 
 ## binarySource
 
-Set this to `global` if you wish Renovate to use globally-installed binaries (`npm`, `yarn`, etc) instead of using its bundled versions.
-Set this to `docker` instead to use Docker-based binaries.
+By default Renovate uses the binaries that are bundled with the Renovate bot.
+Use the `binarySource` option to override this default.
+You can use globally installed binaries, or Docker-based binaries.
+
+To use your globally installed binaries (`npm`, `yarn`, etc):
+
+```json
+{
+  "binarySource": "global"
+}
+```
+
+To use Docker-based binaries:
+
+```json
+{
+  "binarySource": "docker"
+}
+```
 
 ## cacheDir
 
-Configure this directory if you want to change which directory Renovate uses for storing cache data.
-If left unconfigured, it will typically be a temporary directory like `/tmp/renovate/cache/`.
-If you configure this to be different to the `baseDir`, it means you can have one location for repo data and another for cache data.
+By default Renovate uses a temporary directory like `/tmp/renovate/cache` to store cache data.
+Use the `cacheDir` option to override this default.
+
+The `baseDir` and `cacheDir` option do not need to point to the same directory.
+You can use one directory for the repo data, and another for the the cache data.
+
+e.g.
+
+```json
+{
+  "baseDir": "/my-own-different-temporary-folder",
+  "cacheDir": "/my-own-different-cache-folder"
+}
+```
 
 ## composerIgnorePlatformReqs
 
 Set to `false` to prevent usage of `--ignore-platform-reqs` in the Composer package manager.
 
+e.g.
+
+```json
+{
+  "composerIgnorePlatformReqs": false
+}
+```
```

Comment:
Don't you think such examples are overkill?

Reply:
This is what the Renovate docs say currently:

Enable / disable use of `--ignore-platform-reqs` in the composer package manager.

Name | Value
-- | --
type | boolean
default | true

Set to `false `to prevent usage of `--ignore-platform-reqs` in the Composer package manager.